plugins/modules/job_invocation.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2020 Peter Ondrejka <pondrejk@redhat.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: job_invocation
+short_description: Manage Job Invocations in Foreman
+description:
+  - "Manage Remote Execution Job Invocations"
+author:
+  - "Peter Ondrejka (@pondrejk)"
+options:
+  search_query:
+    description:
+      - Search query to identify hosts
+    type: str
+  bookmark_id:
+    description:
+      - Bookmark ID to infer the search query from
+    choices:
+      - 6 (active)
+      - 9 (disabled)
+      - 8 (error)
+      - 1 (list hypervisors)
+      - 10 (ok hosts)
+      - 7 (out of sync)
```

Comment:
Where do the numbers come from? I guess they are special to your system?
As a rule of thumb, we do not expose any id's in the user facing part of the ansible module.
Instead we ask the user for the name of the bookmark and do the id-lookup internally.
I'll keep the bookmarks as an example, but with job_templates it is the same deal.